File: core/src/main/java/com/sequenceiq/cloudbreak/controller/json/StackValidationRequest.java
package com.sequenceiq.cloudbreak.controller.json;

import com.sequenceiq.cloudbreak.controller.doc.ModelDescriptions.StackModelDescription;
import com.wordnik.swagger.annotations.ApiModel;
import com.wordnik.swagger.annotations.ApiModelProperty;

import java.util.HashSet;
import java.util.Set;

import javax.validation.constraints.NotNull;

@ApiModel
public class StackValidationRequest implements JsonEntity {
    @ApiModelProperty(required = true)
    private Set<HostGroupJson> hostGroups = new HashSet<>();
    @ApiModelProperty(required = true)
    private Set<InstanceGroupJson> instanceGroups = new HashSet<>();
    @NotNull
    @ApiModelProperty(value = StackModelDescription.BLUEPRINT_ID, required = true)
    private Long blueprintId;

    public Set<HostGroupJson> getHostGroups() {
        return hostGroups;
    }

    public void setHostGroups(Set<HostGroupJson> hostGroups) {
        this.hostGroups = hostGroups;
    }

    public Set<InstanceGroupJson> getInstanceGroups() {
        return instanceGroups;
    }

    public void setInstanceGroups(Set<InstanceGroupJson> instanceGroups) {
        this.instanceGroups = instanceGroups;
    }

    public Long getBlueprintId() {
        return blueprintId;
    }

    public void setBlueprintId(Long blueprintId) {
        this.blueprintId = blueprintId;
    }
}

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/model/StackValidationRequest.kt
package com.sequenceiq.cloudbreak.api.model

import java.util.HashSet

import javax.validation.constraints.NotNull

import com.sequenceiq.cloudbreak.doc.ModelDescriptions
import com.sequenceiq.cloudbreak.doc.ModelDescriptions.StackModelDescription

import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty

@ApiModel
class StackValidationRequest : JsonEntity {
    @ApiModelProperty(required = true)
    var hostGroups: Set<HostGroupJson> = HashSet()
    @ApiModelProperty(required = true)
    var instanceGroups: Set<InstanceGroupJson> = HashSet()
    @NotNull
    @ApiModelProperty(value = StackModelDescription.BLUEPRINT_ID, required = true)
    var blueprintId: Long? = null
    @NotNull
    @ApiModelProperty(value = StackModelDescription.NETWORK_ID, required = true)
    var networkId: Long? = null
    @NotNull
    @ApiModelProperty(value = ModelDescriptions.CLOUD_PLATFORM, required = true)
    var platform: String? = null
    var fileSystem: FileSystemRequest? = null
}
